examples/colors/store.rs

```diff
@@ -0,0 +1,193 @@
+use glib::{ParamFlags, ParamSpec, Value};
+use gtk::gdk;
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+
+use crate::color::Color;
+
+const n_colors: usize = 256 * 256 * 256;
+
+mod imp {
+    use std::cell::RefCell;
+
+    use super::*;
+    use gtk::glib::once_cell::sync::Lazy;
+
+    pub struct ColorStore {
+        pub colors: RefCell<Vec<Color>>,
+        pub size: RefCell<u32>,
+        pub limit: RefCell<u32>,
+    }
+
+    impl Default for ColorStore {
+        fn default() -> Self {
+            Self {
+                colors: RefCell::new(vec![Color::new(gtk::gdk::RGBA::RED); n_colors]),
+                size: RefCell::new(0),
+                limit: RefCell::new(0),
+            }
+        }
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for ColorStore {
+        const NAME: &'static str = "ColorStore";
+        type Type = super::ColorStore;
+        type Interfaces = (gtk::gio::ListModel,);
+    }
+
+    impl ObjectImpl for ColorStore {
+        fn constructed(&self, obj: &Self::Type) {
+            let mut iter = include_str!("color.names.txt").split('\n');
+
+            let mut size = 0;
```

Comment:
this whole part of parsing should be moved to a separate function and be called here